### PR TITLE
chore: refactor `chanUpgradeAck` tests to use expected errors

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Follow up from #3828 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/upgrade_test.go [hunk 1]
```diff
@@ -398,46 +398,47 @@ func (suite *KeeperTestSuite) TestChanUpgradeAck() {
...
 			"upgrade not found",
 			func() {
 				store := suite.chainA.GetContext().KVStore(suite.chainA.GetSimApp().GetKey(exported.ModuleName))
 				store.Delete(host.ChannelUpgradeKey(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID))
 			},
-			false,
+			types.ErrUpgradeNotFound,
 		},
 		{
-			"channel end version mismatch on crossing hellos",
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/upgrade_test.go:412]: moved to bottom as this is last in the call stack

File: modules/core/04-channel/keeper/upgrade_test.go [hunk 2]
```diff
@@ -398,46 +398,47 @@ func (suite *KeeperTestSuite) TestChanUpgradeAck() {
...
-
-				path.EndpointA.SetChannelUpgrade(upgrade)
+				counterpartyUpgrade.Fields.ConnectionHops = []string{ibctesting.InvalidID}
 			},
-			false,
+			commitmenttypes.ErrInvalidProof,
 		},
 		{
-			"startFlushUpgradeHandshake fails due to proof verification failure, counterparty upgrade connection hops are tampered with",
+			"startFlushUpgradeHandshake fails due to mismatch in upgrade ordering",
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/upgrade_test.go:419]: I changed this testcase from: `"startFlushUpgradeHandshake fails due to mismatch in upgrade sequences"` to this one which fails with ordering mismatch. 

This was the one testcase which was actually failing on proof verification before we got to the checks afterwards. 
We can't force mismatch on upgrade sequences in this test as the counterparty channel upgrade sequence is built using our channel upgrade sequence, so mutating one or the other causes proof verification to fail. Decided to test for mismatch in upgrade ordering instead

File: modules/core/04-channel/keeper/upgrade.go [hunk 3]
```diff
@@ -204,32 +204,6 @@ func (k Keeper) WriteUpgradeTryChannel(ctx sdk.Context, portID, channelID string
 	return channel, upgrade
 }
 
-// WriteUpgradeAckChannel writes a channel which has successfully passed the UpgradeAck handshake step as well as
-// setting the upgrade for that channel.
-// An event is emitted for the handshake step.
-func (k Keeper) WriteUpgradeAckChannel(
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/upgrade.go:210]: moved this to below `ChanUpgradeAck`

**Review by chatton — APPROVED** [review]
LGTM great improvement

**Review by crodriguezvega — APPROVED** [review]

**Review by charleenfei — APPROVED** [review]

**Pull request merged**